### PR TITLE
Remove all implicit waits and clean up page objects

### PR DESCRIPTION
**Pull request by bobsilverberg**
@mozilla/web-qa-sorcerers r?

### CODE REVIEW

**Comment by bobsilverberg** [comment]
Adhoc on Jenkins stage is in the queue

**Comment by davehunt** [comment]
Adhoc: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/oneanddone.adhoc/42/

File: pages/base.py [hunk 1]
```diff
@@ -13,49 +14,43 @@ class Base(Page):
...
+    def is_user_logged_in(self):
+        return not self.is_element_visible(self._browserid_login_locator)
 
     @property
-    def is_user_logged_in(self):
-        return self.is_element_not_visible(*self._browserid_login_locator)
+    def profile_link_text(self):
+        return self.selenium.find_element(*self._profile_link_locator).text
+
+    def click_browserid_login(self):
```
Comment (davehunt) [inline pages/base.py]: Nit: Is there another login link to need to distinguish?

File: pages/page.py [hunk 2]
```diff
@@ -2,126 +2,68 @@
...
 
-    def is_element_present(self, *locator):
-        self.selenium.implicitly_wait(0)
+    def is_element_visible(self, locator):
         try:
-            self._selenium_root.find_element(*locator)
-            return True
-        except NoSuchElementException:
+            return self.selenium.find_element(*locator).is_displayed()
+        except (NoSuchElementException,):
```
Comment (davehunt) [inline pages/page.py]: Nit: No need for the `,`

**Comment by davehunt** [comment]
Nice improvements! All comments are nits. r+

File: pages/page.py (hunk 2 above)
Comment (bobsilverberg) [inline pages/page.py]: Well, I believe the parentheses are there because this is a tuple, and I also believe that a tuple needs the comma, so I think this is actually more correct than just `(NoSuchElementException)` which Python actually interprets as just a variable inside parens. It would probably be better to do away with the parens entirely as they are what is unnecessary, but I also know that without them there, if the time comes when we need to add another exception to the list, we would likely add it without the parens and that would be wrong.

So, to sum up, we don't need the parens and they should be removed, *but* removing them increases the chance of a bug being introduced later if we want to add another exception, *therefore* if we are going to keep them we should code it correctly as a tuple which means adding the comma.

What do you think?

File: pages/base.py (hunk 1 above)
Comment (bobsilverberg) [inline pages/base.py]: No, there isn't. I'll clean this up.
